### PR TITLE
[no-sq] Minor networking code cleanup

### PR DESCRIPTION
**Pull request by red-001**
Some small issues I fixed while working on a socket-free implementation of simple singleplayer mode.
`TryReceive` and `ReceiveTimeoutMs` should probably be merged into one function, but I didn't do that cause I was focusing on the singleplayer mode changes.

### CODE REVIEW

**Review by sfan5 — APPROVED** [review]
lgtm

File: src/server.cpp [hunk 1]
```diff
@@ -1069,19 +1069,19 @@ void Server::Receive(float timeout)
 		pkt.clear();
 		peer_id = 0;
 		try {
-			if (!m_con->ReceiveTimeoutMs(&pkt,
+			if (m_con->ReceiveTimeoutMs(&pkt,
```
Comment (SmallJoker) [inline src/server.cpp:1072]: IMO a more intuitive change would be to use `continue;` if the time condition on L1082 does not apply. Early guards are commonly easier to read than nested code.

File: src/unittest/test_connection.cpp [hunk 2]
```diff
@@ -338,14 +335,13 @@ void TestConnection::testConnectSendReceive()
 		for (;;) {
 			if (porting::getTimeMs() - timems0 > 5000 || received)
 				break;
-			try {
-				NetworkPacket pkt;
-				client.Receive(&pkt);
+			NetworkPacket pkt;
+			if (client.ReceiveTimeoutMs(&pkt, timeout_ms))
+			{
```
Comment (SmallJoker) [inline src/unittest/test_connection.cpp:340]: At least new code should follow the guidelines.
```suggestion
			if (client.ReceiveTimeoutMs(&pkt, timeout_ms)) {
```

**Review by SmallJoker — COMMENTED** [review]
Minor comments. The unittests pass, so that's a good sign.

File: src/server.cpp (hunk 1 above)
Comment (red-001) [reply src/server.cpp:1072]: depends on code style, but you're that Minetest largely follows the early exit style. I'll change it.

**Review by SmallJoker — APPROVED** [review]
Looks good. Thank you.

**Pull request merged**